torch_xla/csrc/helpers.h

```diff
@@ -48,6 +49,17 @@ class XlaHelpers {
     return xla::ConstantLiteral(builder, ScalarLiteral(scalar_value, type));
   }
 
+  static xla::XlaOp ScalarValue(const at::Scalar& scalar_value,
+                                xla::PrimitiveType type,
+                                xla::XlaBuilder* builder) {
+    if (scalar_value.isFloatingPoint()) {
+      return ScalarValue(scalar_value.toDouble(), type, builder);
+    }
+    XLA_CHECK(scalar_value.isIntegral()) << "Scalar type not supported";
```

Comment:
Stream the scalar as well is at::Scalar supports it.

Reply:
Only for integral and double, not complex (for example).